mantidimaging/gui/windows/spectrum_viewer/presenter.py

```diff
@@ -565,19 +565,75 @@ def setup_fitting_model(self) -> None:
         init_params = self.model.fitting_engine.get_init_params_from_roi(fitting_region)
         self.view.scalable_roi_widget.set_parameter_values(init_params)
+
+        self.view.fittingDisplayWidget.set_plot_mode("initial")
+
         self.show_initial_fit()
         roi_name = self.view.roiSelectionWidget.current_roi_name
         self.view.exportDataTableWidget.update_roi_data(roi_name=roi_name, params=init_params, status="Initial")
 
-    def show_initial_fit(self):
+    def _plot_initial_fit(self) -> None:
+        assert self.model.tof_data is not None
         init_params = self.view.scalable_roi_widget.get_initial_param_values()
         xvals = self.model.tof_data
         init_fit = self.model.fitting_engine.model.evaluate(xvals, init_params)
-        self.view.fittingDisplayWidget.show_init_fit(xvals, init_fit)
+        self.view.fittingDisplayWidget.show_fit_line(xvals,
+                                                     init_fit,
+                                                     color=(128, 128, 128),
+                                                     label="initial",
+                                                     initial=True)
+
+    def _plot_fitted_curve(self) -> None:
+        assert self.model.tof_data is not None
+        fitted_params = self.view.scalable_roi_widget.get_fitted_param_values()
+        xvals = self.model.tof_data
+        fit = self.model.fitting_engine.model.evaluate(xvals, fitted_params)
+        self.view.fittingDisplayWidget.show_fit_line(xvals, fit, color=(0, 128, 255), label="fit", initial=False)
+
+    def on_initial_params_edited(self) -> None:
+        """
+        Handles updates when the initial fitting parameters are edited.
+
+        If the initial fit is visible, updates the plot with the new initial fit.
+        Otherwise, re-runs the fit with the updated parameters, updates the fitted parameter values,
+        and displays the new fit result.
+        """
+        assert self.model.tof_data is not None
+        if self.view.fittingDisplayWidget.is_initial_fit_visible():
+            self._plot_initial_fit()
+        else:
+            init_params = self.view.scalable_roi_widget.get_initial_param_values()
+            roi_name = self.view.roiSelectionWidget.current_roi_name
+            roi = self.view.spectrum_widget.get_roi(roi_name)
+            spectrum = self.model.get_spectrum(roi, self.spectrum_mode)
+            xvals = self.model.tof_data
+            result = self.model.fitting_engine.find_best_fit(xvals, spectrum, init_params)
+            self.view.scalable_roi_widget.set_fitted_parameter_values(result)
+            self.show_fit(list(result.values()))
+
+    def show_initial_fit(self) -> None:
+        """
+        Display either the initial fit or the fitted curve on the fitting display,
+        depending on whether the initial fit is currently visible.
+        """
+        assert self.model.tof_data is not None
+        xvals = self.model.tof_data
```

Comment:
This is called when the "From ROI" button is pressed, so it should always be updating the initial and showing an initial curve. So it does not need the `if/else`.
(I don't think it can actually hit the else case currently).

Reply:
I think the `if` can also be removed, and the docstring needs update